workflow/artifacts/gcs/gcs.go

```diff
@@ -177,6 +177,47 @@ func downloadObject(client *storage.Client, bucket, key, objName, path string) e
 	return nil
 }
 
+func (g *ArtifactDriver) Delete(artifact *wfv1.Artifact) error {
```

Comment:
What is the use-case for deleting the object?

Reply:
> When a Workflow is deleted, we should consider garbage collecting artifacts from those workflows where possible. This should likely be configurable. 

As described in #3102 

This feature would require the ability to delete artifacts. And this is the reason for this function.